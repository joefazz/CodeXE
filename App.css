.App {
    text-align: center;
}

.App-logo {
    animation: App-logo-spin infinite 40s linear;
    height: 40vmin;
    top: 19%;
    right: 25%;
    position: absolute;
    z-index: 100;
    justify-self: flex-start;
    align-self: flex-start;
}

.App-header {
    background-color: #282c34;
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    font-size: calc(10px + 2vmin);
    color: white;
}

.App-link {
    color: #61dafb;
}

a {
    color: dodgerblue;
}

@keyframes App-logo-spin {
    from {
        transform: rotate(0deg);
    }
    to {
        transform: rotate(360deg);
    }
}
