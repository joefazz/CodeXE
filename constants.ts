import { Languages } from './@types';

export const colors = {
    mainBlue: 'dodgerblue',
    mainGreen: '#3ca56e',
    backgroundDark: '#282c34',
    backgroundBlue: '#2d3e5d',
    backgroundDarkTranslucent: 'rgba(40,44,52, 0.7)'
};

export const fonts = {
    display: "'Arvo', serif",
    body: "'Josefin Sans', sans-serif"
};

export const languageOptions: { value: string; label: string }[] = [
    { value: Languages.JS, label: 'JavaScript' },
    { value: Languages.PYTHON, label: 'Python' },
    { value: Languages.C, label: 'C/C++' }
];

export const DOMAIN = `https://midgard.codexe.run`;
