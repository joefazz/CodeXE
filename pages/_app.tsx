import App, { Container, NextAppContext } from 'next/app';
import Head from 'next/head';
import React from 'react';
import { MessageTypes } from '../@types';
import { handleMessage } from '../functions/websockets';

export const SocketContext = React.createContext({});

export default class MyApp extends App {
    containerInfo: any;
    socket: WebSocket;

    static async getInitialProps({ Component, ctx }: NextAppContext) {
        let pageProps = {};

        if (Component.getInitialProps) {
            pageProps = await Component.getInitialProps(ctx);
        }

        return { pageProps };
    }

    state = {
        status: 'disconnected',
        containerName: '',
        id: '',
        activityId: '',
        response: {
            readData: {},
            metaData: { saveInfo: {} },
            writeData: { output: 'Output' }
        }
    };

    componentDidMount() {
        this.socket = new WebSocket('ws://localhost:4000/');

        this.socket.onopen = () => {
            console.log('Socket Opened');
        };

        this.socket.onmessage = (event) => this.setState(handleMessage(event, this.state));

        this.socket.onclose = function() {
            // socket.send('container.end');
            console.log('WebSocket has been closed');
        };
        this.socket.onerror = function(event) {
            console.log('Error', event);
        };
    }

    // TODO: GET THIS WORKING FFS
    componentWillUnmount() {
        this.socket.send(
            JSON.stringify({ message: 'Container.Stop', data: { id: this.state.id } })
        );
        this.socket.close();
    }

    render() {
        const { Component, pageProps } = this.props;
        return (
            <Container>
                <Head>
                    <title>CodeXE</title>
                    <link
                        href="https://fonts.googleapis.com/css?family=Cantarell|Arvo"
                        rel="stylesheet"
                    />
                    <link
                        rel="shortcut icon"
                        href="/static/images/favicon.ico"
                        type="image/x-icon"
                    />
                    <link rel="icon" href="/static/images/favicon.ico" type="image/x-icon" />
                    <link
                        rel="stylesheet"
                        href="https://cdnjs.cloudflare.com/ajax/libs/normalize/8.0.0/normalize.min.css"
                    />
                </Head>
                <SocketContext.Provider value={{ ...this.state, socket: this.socket }}>
                    <Component {...pageProps} />
                </SocketContext.Provider>
            </Container>
        );
    }
}
