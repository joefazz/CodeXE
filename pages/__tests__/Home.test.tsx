/* eslint-env: jest */

import React from 'react';

// @ts-ignore
import preloadAll from 'jest-next-dynamic';

import App from '..';
import { render } from 'react-testing-library';
import { SocketContext } from '../_app';
import { CONTEXT_DEFAULT } from '../../constants';

// describe('Renders the code editor', () => {
//     it('Shows the code editor', async () => {
//         const { getByTestId } = render(
//             <SocketContext.Provider value={CONTEXT_DEFAULT}>
//                 <SocketContext.Consumer>{() => <App />}</SocketContext.Consumer>
//             </SocketContext.Provider>
//         );

//         await waitForElement(() => getByTestId('monaco'), { timeout: 1000 });

//         expect(getByTestId('monaco')).not.toBeNull();
//     });
// });

// describe('Code Editor Tests', () => {
//     it('Renders', async () => {
//         const { getByTestId } = render(<Monaco />);

//         expect(getByTestId('monaco')).not.toBeNull();
//     });
// });

describe('Home Page Snapshot', () => {
    it('Should match Snapshot', () => {
        const { asFragment } = render(
            <SocketContext.Provider value={CONTEXT_DEFAULT}>
                <SocketContext.Consumer>{() => <App />}</SocketContext.Consumer>
            </SocketContext.Provider>
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
