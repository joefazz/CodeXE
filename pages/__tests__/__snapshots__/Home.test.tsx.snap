// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Snapshot Should match Snapshot 1`] = `
<DocumentFragment>
  <div
    class="Layout__RootPage-sc-19ikuwz-0 jmOFgE"
  >
    <header
      class="Header-sc-2vudz7-0 iCnnCf"
    >
      <div
        class="Layout__LeftWrapper-sc-19ikuwz-1 jJmoFf"
      >
        <a
          class="Logo-sc-108q7dg-0 hhrpgI"
        >
          CodeXE
        </a>
        <div
          class="Layout__StatusWrapper-sc-19ikuwz-2 bjSJPB"
        >
          <span
            style="padding-bottom: 2px;"
          >
            Connection Status:
          </span>
           
          <div
            class="Layout__ContainerStatus-sc-19ikuwz-3 idJFII"
            color="red"
          />
        </div>
      </div>
      <div
        style="display: flex; justify-content: space-evenly; align-items: center; width: 40%;"
      >
        <a
          class="NavLink-sc-1yp13yj-0 fHJUjJ"
        >
          Sandbox
        </a>
        <a
          class="NavLink-sc-1yp13yj-0 fHJUjJ"
        >
          Exercises
        </a>
      </div>
    </header>
    <div
      style="background-color: rgb(45, 62, 93); padding-top: 75px; width: 100vw;"
    >
      <div
        class="home__Page-ahlzao-0 jCJZlP"
      >
        <div
          class="home__CodeSection-ahlzao-2 kusRdq"
        >
          <h1
            class="home__H1-ahlzao-7 guLOiD"
          >
            Do coding exercises right in your browser!
          </h1>
          <ul
            class="home__LanguageWrapper-ahlzao-6 fQDkWg"
          >
            <li
              style="border-top-left-radius: 5px; border-bottom-left-radius: 5px;"
            >
              JavaScript
            </li>
            <li>
              Python
            </li>
            <li
              style="border-top-right-radius: 5px; border-bottom-right-radius: 5px;"
            >
              C
            </li>
          </ul>
          <div
            style="height: 53.2%;"
          >
            <div
              class="LoadingCode__CodeLoad-sc-1cr3tz2-0 dPxXKw"
            >
              <img
                class="LoadingCode__VSLoader-sc-1cr3tz2-1 jFvteS"
                src="/static/images/vscode-b&w.png"
              />
              <h1
                class="H1-sc-1otcw4s-0 krdItu"
              >
                Loading Code...
              </h1>
            </div>
          </div>
          <div>
            <button
              class="Button-sc-1w81v9u-0 gqAGCh"
              style="font-size: 1.8rem; border-top-left-radius: 0; border-top-right-radius: 0;"
            >
              Run
            </button>
          </div>
          <span
            class="CodeOutput-fl4bg6-0 fnbzUW"
          >
            Disconnected
          </span>
        </div>
        <div
          class="home__InfoSection-ahlzao-3 cmUEBH"
        >
          <img
            alt="logo"
            class="App-logo"
            src="/static/svgs/Moon.svg"
          />
          <ul
            class="home__FeatureArea-ahlzao-4 kSuieO"
          >
            <div
              style="margin-top: 10%; shape-outside: circle(40%); height: 300px; width: 35vh; max-height: 300px; max-width: 300px; float: left; text-align: right;"
            />
            <br />
            <li>
              <img
                alt="li"
                class="home__EmojiListIcon-ahlzao-1 igxaRr"
                src="/static/images/earth-emoji.png"
              />
              Automagically generated exercises
            </li>
            <li>
              <img
                alt="li"
                class="home__EmojiListIcon-ahlzao-1 igxaRr"
                src="/static/images/earth-emoji.png"
              />
              Fully featured code editor
            </li>
            <li>
              <img
                alt="li"
                class="home__EmojiListIcon-ahlzao-1 igxaRr"
                src="/static/images/earth-emoji.png"
              />
              Interact with a full Linux Virtual Machine
            </li>
            <li>
              <img
                alt="li"
                class="home__EmojiListIcon-ahlzao-1 igxaRr"
                src="/static/images/earth-emoji.png"
              />
              Learn the basics of some of the most popular languages
            </li>
            <li>
              <img
                alt="li"
                class="home__EmojiListIcon-ahlzao-1 igxaRr"
                src="/static/images/earth-emoji.png"
              />
              Make your own exercise and share it
            </li>
            <li>
              <img
                alt="li"
                class="home__EmojiListIcon-ahlzao-1 igxaRr"
                src="/static/images/earth-emoji.png"
              />
              No set up required
            </li>
          </ul>
        </div>
        <div
          class="home__ButtonArea-ahlzao-5 fKdXFG"
        >
          <button
            class="Button-sc-1w81v9u-0 ILFva"
            style="box-shadow: 0 0 20px black;"
          >
            Log In
          </button>
          <button
            class="Button-sc-1w81v9u-0 ebJCPF"
            style="border-top-right-radius: 0; border-bottom-right-radius: 0; box-shadow: 0 0 20px black;"
          >
            Sign Up
          </button>
        </div>
        <div
          class="Footer-sc-19yne8k-0 jBOzTc"
        >
          <p>
            CodeXE is the final year project of 
            <a
              href="https://twitter.com/purefazz"
            >
              Joe Fazzino
            </a>
            , it lowers the barrier to entry for development by giving everyone an online environment!
          </p>
          <p>
            <code>
              Pronounced: Codex; Code-Ex-Eee; or CodeXECUTE
            </code>
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
